content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -1,48 +1,75 @@
+
+- The `Socket` class constructor will attempt to connect the client to the server and port number provided. The client now has a socket object that can connect to the server if communication occurs.
+
+- The server's accept() method returns a reference to a new server socket connected to the client's socket.
  
-If the connection is successful, the client and server can then communicate using **I/O streams**. The **I/O streams** come from socket objects of the client and server. The client's `OutputStream` connects to the server's `InputStream` while the server's `OutputStream` connects to the client's `InputStream`.
-In case you don't know, a stream is a collection of sequenced data. They can be divided into two categories:
+The client and server can then communicate using **I/O streams** If the connection is successful. The client and server socket classes provide the **I/O streams**. The client's OutputStream communicates with the server's InputStream, and the server's OutputStream communicates with the client's InputStream.
+A stream is a collection of sequenced data, in case you didn't know. 
+
+There are two types of streams:
 1. Character stream (usually used with text files)
 2. Byte stream (usually used with images)
+
 Character stream is in human-readable language while byte stream is in machine language, a language understandable by the CPU.
-In our case, the client and server will send messages to each other's streams in parallel. This is made possible because TCP is a two-way communication protocol.
-   
-### Server-side Programming 
-A server socket is an instance of the `ServerSocket` class and can be created by one of the following constructors: 
+
+In this situation, the client and server will simultaneously broadcast messages to each other's streams. Because TCP is a two-way communication protocol, this is conceivable.
+
+#### Programming on the Server-side
+
+A `server` socket is a subclass of the ServerSocket that uses the constructor below:
+
 ```java
-serversocket = new Serversocket (int port)
+serversocket = new ServerSocket (int port)
 ```
-port: port number at which the server will be listening for requests from clients.
-The ServerSocket object sits and waits for network requests to arrive. Our client will try to connect to port number 5000 that we selected. 
-**Note** Make sure the ports are the same otherwise there will be no connection
-Two while loops will be in use: 
-**First Loop**- This is to ensure the server is constantly running.
-**Second Loop**- Ensures that once the client is connected, the server interacts with the client until the client disconnects.
+Port: the port number on which the server will listen for the client's requests.
+
+The ServerSocket object stays idle, waiting for network requests. Our client will attempt to connect to port 5000, which we have specified.
+
+**Note** Make sure the ports are the same other than that, the connection will fail.
```

Comment:
> **Note** Make sure the ports are the same other than that, the connection will fail.

Put this in a blockquote, add a **:** after **Note**, and put a new line after like so:

> \> **Note:** Make sure the ports are the same other than that, the connection will fail.
>
>There will be two while loops in use: